handlers/stripe-disputes/src/producer.ts

```diff
@@ -23,22 +45,55 @@ export const handler = async (
 ): Promise<APIGatewayProxyResult | void> => {
 	logger.log(`Input: ${JSON.stringify(event)}`);
 
-	if (isApiGatewayEvent(event)) {
+	const stripeSignature: string | undefined = event.headers['Stripe-Signature'];
+
+	logger.log(`Headers: ${JSON.stringify(event.headers)}`);
+	logger.log(
+		`Stripe-Signature: ${JSON.stringify(event.headers['Stripe-Signature'])}`,
+	);
+
+	if (!stripeSignature) {
+		logger.error('Missing Stripe-Signature header');
+		return {
+			statusCode: 400,
+			body: JSON.stringify({ message: 'Missing Stripe-Signature header' }),
+		};
+	}
+
+	if (!event.body) {
+		logger.error('Missing request body');
+		return {
+			statusCode: 400,
+			body: JSON.stringify({ message: 'Missing request body' }),
+		};
+	}
+
+	const endpointSecretObject: StripeCredentials =
+		await getSecretValue<StripeCredentials>(
+			`${stageFromEnvironment()}/Stripe/ConnectedApp/StripeDisputeWebhooks`,
+		);
+
+	logger.log({
+		webhook_endpoint_secret: endpointSecretObject.webhook_endpoint_secret,
+		secret_key: endpointSecretObject.secret_key,
+	});
```

Comment:
Please avoid logging secrets unless really necessary:

```suggestion
	logger.log({
		webhook_endpoint_secret: endpointSecretObject.webhook_endpoint_secret,
		secret_key: endpointSecretObject.secret_key,
	});
```

Reply:
Not sure if the `Stripe-Signature` header is sensitive and we should also avoid logging it. I think it's probably fine, but we might remove it to be safe.

Reply:
Ho! sorry i was for debuging ! Thanks having seen ! @graham228221